pkg/util/workqueue/prometheus/prometheus.go

```diff
@@ -37,7 +38,9 @@ func (prometheusMetricsProvider) NewDepthMetric(name string) workqueue.GaugeMetr
 		Name:      "depth",
 		Help:      "Current depth of workqueue: " + name,
 	})
-	prometheus.Register(depth)
+	if err := prometheus.Register(depth); err != nil {
+		klog.Errorf("failed to register depth metric %v: %v", name, err)
```

Comment:
shouldn't this me a more fatal error actually? It's a programming error. Those names are static strings, i.e. they don't come from the user.

Reply:
there are more failures caused by duplication that I didn't dig into (see the [gist](https://gist.github.com/roycaihw/1bbb77d38e198b2528651b660da6c9da)). We can have a followup issue to clean those up

Reply:
it's not a complete list. kcm/ccm may also have such failure

Reply:
created #73551